src/components/TileEditor.tsx

```diff
@@ -1,14 +1,78 @@
 import React from 'react';
-import { TileInfo } from '../firebase';
+import { firestore, TileInfo } from '../firebase';
+import { useForm } from "react-hook-form";
 
 export function TileEditor(props: {tileInfo: TileInfo, tileName: string}) {
+  const { register, setValue, handleSubmit, errors } = useForm<TileInfo>();
+  const onSubmit = handleSubmit( async ({ bg_color, bg_image, hover_text, description_text, link }) => {
```

Comment:
Could we move this function to firebase.tsx? I'd like to keep all the functions that manipulate the db in one place.

Reply:
Makes sense to me - moved